### PR TITLE
feat: remove length restriction

### PR DESCRIPTION
**Pull request by dignifiedquire**
Not sure if we need to do anything else here. 

@diasdavid @jbenet please check

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -162,14 +162,6 @@ describe('mh', () => {
...
         /length should be equal/
       )
-
-      const longBuffer = new Buffer(150)
-      longBuffer.fill('a')
-      expect(
-        () => mh.encode(longBuffer, 0x11)
-      ).to.throw(
-        /not yet support/
-      )
```
Comment (diasdavid) [inline test/index.spec.js]: add a test that checks something that is added with more than 127 bytes and that the varint added is correct.

File: src/index.js [hunk 2]
```diff
@@ -69,10 +69,6 @@ exports.encode = function encode (digest, code, length) {
     throw new Error('digest length should be equal to specified length.')
   }
 
-  if (length > 127) {
```
Comment (dignifiedquire) [inline src/index.js]: It's currently not using a varint, but I'm unclear from the multihash spec what exactly to use as this length uses a fixed size of 4 bytes.

**Comment by dignifiedquire** [comment]
nevermind, don't need this actually :)

File: src/index.js (hunk 2 above)
Comment (diasdavid) [inline src/index.js]: Correction, in a full multihash buffer there is a:

`<multicodec varint><varint digest length><digest>`

See https://github.com/multiformats/multihash#format for reference

**Comment by diasdavid** [comment]
@dignifiedquire doesn't js-multihashing use js-multihash?

**Pull request closed without merging**